frontend/src/metabase/lib/core.ts

```diff
@@ -2,13 +2,15 @@ import { t } from "ttag";
 
 import type { IconName } from "metabase/ui";
 import { TYPE } from "metabase-lib/v1/types/constants";
+import type { Field } from "metabase-types/api";
 
 interface FieldSemanticType {
-  id: string;
+  id: NonNullable<Field["semantic_type"]>;
```

Comment:
This type just resolves to `string` when I open this branch locally. So `id: "foo"` also works which is probably not what you want.

Should it be more restrictive like `keyof typeof TYPE`?

```suggestion
  id: keyof typeof TYPE;
``` 

Maybe that type could even get a name?

Reply:
> This type just resolves to string when I open this branch locally.

I did it to indicate a connection between the types.
I don't expect this type to be more precise than `string` in any foreseeable future, especially because (see below)

> Should it be more restrictive like `keyof typeof TYPE`?

`TYPE` is imported from cljs, it's not typed at all